### DISCUSSION TITLE
Does "IN" is reserved keyword in redis?

### DISCUSSION THREAD

**Discussion by hemangi225** [xetorthio/jedis#3419] [Q&A]
Does IN is reserved keyword in redis? Because in my case, when json data contains string which is starting with “IN” and also contains special character in it (e.g. IN-hemangiblpt) then there is no need to escape that special character during search. But if string started other than IN and contains special character then i need to escape it in data aswell as in query.

This is my redis data which contains 2 JSON keys with different name.

1.	Key is “AgentData:90009f26-b05b-4d87-99c0-db7290db081a”

JsonValue is 

{
name:"IN-hemangiblpt"
ip:"192.168.31.144"
token:"90009f26-b05b-4d87-99c0-db7290db081a"
isAliveTill:"2023-05-16T10:09:40.011Z"
}

2.	Key is “AgentData:af1a9aee-c623-4316-90a0-c443a78ebf56”

JsonValue is

{
name:"hemangib-lpt"
ip:"192.168.31.145"
token:"af1a9aee-c623-4316-90a0-c443a78ebf56"
isAliveTill:"2023-05-16T10:08:20.013Z"
}

Index on redis for AgentData  :
1) "index_name" 2) "idx:AgentData" 3) "index_options" 4) (empty list or set) 5) "index_definition" 6) 1) "key_type" 2) "JSON" 3) "prefixes" 4) 1) "AgentData" 5) "default_score" 6) "1" 
7) "attributes"
 1) 1) "identifier" 2) "$.name" 3) "attribute" 4) "name" 5) "type" 6) "TEXT" 7) "WEIGHT" 8) "1"
 2) 1) "identifier" 2) "$.ip" 3) "attribute" 4) "ip" 5) "type" 6) "TEXT" 7) "WEIGHT" 8) "1"
 3) 1) "identifier" 2) "$.token" 3) "attribute" 4) "token" 5) "type" 6) "TEXT" 7) "WEIGHT" 8) "1"

Search Result for all AgentData using query as *
> ft.search "idx:AgentData" * 
1) "2"
2) "AgentData:af1a9aee-c623-4316-90a0-c443a78ebf56"
3) 1) "$" 2) "{\"name\":\"hemangib-lpt\",\"ip\":\"192.168.31.145\",\"token\":\"af1a9aee-c623-4316-90a0-c443a78ebf56\",\"isAliveTill\":\"2023-05-16T10:08:20.013Z\"}"
4) "AgentData:90009f26-b05b-4d87-99c0-db7290db081a" 
5) 1) "$" 2) "{\"name\":\"IN-hemangiblpt\",\"ip\":\"192.168.31.144\",\"token\":\"90009f26-b05b-4d87-99c0-db7290db081a\",\"isAliveTill\":\"2023-05-16T10:17:00.008Z\"}"		

**When name field starts with IN, and if it contains characters like -, there is no need to escape it during search. We get correct result** like
Query: > ft.search "idx:AgentData" "@name:IN-hemangiblpt" 
Result:  1) "1" 2) "AgentData:af1a9aee-c623-4316-90a0-c443a78ebf56" 3) 1) "$" 2) "{\"name\":\"hemangib-lpt\",\"ip\":\"192.168.31.145\",\"token\":\"af1a9aee-c623-4316-90a0-c443a78ebf56\",\"isAliveTill\":\"2023-05-16T10:08:20.013Z\"}"

**But if we do search on 2nd data where name string doesn’t starts with IN but contains - in it, then search result failed to return data.**
Query : > ft.search "idx:AgentData" "@name:hemangib-lpt" 
Result:  1) "0"

**If we need to search second record, then we need to escape that character in redis db and need to pass query also with escape string** like
Data in redis with escape string:
{
name:"hemangib\-lpt"
ip:"192.168.31.145"
token:"af1a9aee-c623-4316-90a0-c443a78ebf56"
isAliveTill:"2023-05-16T10:08:20.013Z"
}

Query : > ft.search "idx:AgentData" "@name:hemangib\\-lpt" 
Result : 1) "1" 2) "AgentData:af1a9aee-c623-4316-90a0-c443a78ebf56" 3) 1) "$" 2) "{\"name\":\"hemangib\\\\-lpt\",\"ip\":\"192.168.31.145\",\"token\":\"af1a9aee-c623-4316-90a0-c443a78ebf56\",\"isAliveTill\":\"2023-05-16T10:08:20.013Z\"}

**Comment by sazzad16** [comment]
Please ask in [RediSearch](https://github.com/RediSearch/RediSearch) or [RedisJSON](https://github.com/RedisJSON/RedisJSON)

**Comment by hemangi225** [comment]
ok. Thanks
